package.json

```diff
@@ -28,6 +28,7 @@
     "start-main-dev": "cross-env HOT=1 NODE_ENV=development electron -r @babel/register ./app/main.dev.js",
     "prepare-qa-build": "node -r @babel/register scripts/prepare-qa-build.js",
     "package": "npm run build-quiet && electron-builder -p always",
+    "publish-s3": "electron-builder -p always -c.publish.provider=s3 -c.publish.bucket=downloads.tidepool.org",
```

Comment:
I tried adding it in add9c6a, but GitHub and S3 binaries still have slightly different sizes.

Reply:
In 3fab96f I put the `publish` object back into `package.json` to see if they have the same size then, and they do. Any idea how I could publish to both github and s3 using the CLI options? 
`-c publish=github -c.publish.provider=s3 -c.publish.bucket=downloads.tidepool.org` doesn't work.

Reply:
So it looks like we need to get `c.publish` to be an `array`, and with `yargs`, the way to do that is to provide two values for the same argument key. I'm not really seeing a way to get it to treat sub-arguments of a potential array argument as properties on an object. If we leave the `publish` in the `package.json` to specify both publish targets, can we override via CLI arguments `-c.publish=github` on our normal `package` command and then have the `publish-s3` command just be `electron-builder -p always`? It's not the most intuitive, but it _might_ get us the result we want?

Reply:
OK, after lots of experimenting it turns out electron-builder merges the CLI config with the `package.json` config, so e.g. if you have

```
"publish": [
      {
        "provider": "s3",
        "bucket": "downloads.tidepool.org"
      },
      {
        "provider": "github"
      }
    ]
```

and you use

     electron-builder -p always -c.publish.provider=github

you get 

```
publish:
                          - provider: github
                            bucket: downloads.tidepool.org
                          - provider: github
```

What happens then is that it attempts to publish to github twice, resulting in github throwing an error message. If you put github first in `package.json`, it will still build for both if you use the CLI config above, as s3 is not overwritten.

Something to keep in mind is that github has to be first in the list, as the first provider is used for auto-update.

Reply:
I guess part of the problem is https://github.com/yargs/yargs/issues/1336 ?

Reply:
Yeah. Maybe we can just make a second `electron-builder.json` file _just_ for the S3 publish and then use the `--config <file>` syntax to point to the entire file? It ends up probably duplicating a lot of the build information, which is against everything that DRY stands for, but I feel like we're exhausting other options at this point.

Reply:
Wait.. what? Got it working, but the file size are still different?! It looks like it has to do with the GitHub binaries not being overwritten after they've been tagged, so running the same build twice can provide different sized binaries?

In any event, that means we can probably go back to 7262d53 and accept that the binaries will have slightly different sizes?

Reply:
Ah, right - I'm betting that since the `publish` influences the auto-update behavior built in to the packaged version of the app, if we have different `publish` settings for one build vs. the other, then the binaries are going to be different.

What if we moved the config into a `.js` file that is an
>  (exported configuration or function that produces configuration)

as a function which looks at `process.env` for the relevant environment variables and then either sets the `publish` to just github or github+s3 based on that?

Reply:
OK, I moved everything into a `.js` file and the file sizes are still different. I think this is because the GitHub binaries can't be overwritten after the tag is created, so that final tagged build get created on S3 but GitHub is skipped. See e.g. https://app.circleci.com/pipelines/github/tidepool-org/uploader/1093/workflows/ab4356f1-870d-41db-aff6-99cae49d17ac/jobs/5434

Reply:
I think they can be overwritten, but the release type needs to match. The publisher seems to be attempting to push to a `draft` release, where the actual Github release with the tag exists as a `pre-release`, so it won't overwrite the artifacts. I think if it's published as a `pre-release`, then builder will overwrite the artifacts.